projects/common/Utils/DebugHandler.h

```diff
@@ -23,7 +23,10 @@
 */
 #pragma once
 
+#ifdef _WIN32
```

Comment:
I think we need to figure out a way to write a more cross-platform DebugHandler. Maybe there are some libraries we can take advantage of to sort most of this out for us? spdlog maybe?

Reply:
We don't need a library for this, the functionality of the DebugHandler is really simple.
First we need to figure out variadric templates and how to pass them along to printf on Unix, this would be the "Args" part of the templates. If I followed the discord conversation correctly this works a bit differently in gcc/clang. Then we need to replace the Windows specific "SetConsoleTextAttribute" with whatever the Unix equivalent is within simple ifdefs.